dashboards/app/pages/data.py

```diff
@@ -168,313 +229,355 @@ def frenchify(x: int) -> str:
+            if len(df_typemilieu) != 0:
+
+                # Afficher le graphique
+                st.plotly_chart(fig3, use_container_width=True)
+
+                # Ne pas faire apparaître la catégorie "Multi-lieux"
+                lignes_multi = df_nb_par_milieu.loc[
+                    df_nb_par_milieu.index == "Multi-lieux"
+                ]
+                df_nb_par_milieu.drop(lignes_multi.index, axis=0, inplace=True)
+
+                # Renommage des colonnes pour l'affichage
+                df_nb_par_milieu.rename(
+                    {
+                        "TYPE_MILIEU": "Milieu",
+                        "ID_RELEVE": "Ramassages",
+                    },
+                    axis=1,
+                    inplace=True,
+                )
+
+                # Convertir en int pour éviter les virgules à l'affichage
+                df_nb_par_milieu = df_nb_par_milieu.astype("int")
+
+                # Affichage du tableau
+                st.write("**Nombre de ramassages par milieu**")
+                st.table(df_nb_par_milieu.T)
+                st.caption(
+                    f"Les ramassages catégorisés en 'Multi-lieux' "
+                    + f"ont été retirés de l'analyse. "
+                    + f"Les milieux représentant moins de 3 ramassages ne sont pas affichés."
+                )
+
+            else:
+                st.warning(
+                    "⚠️ Aucune donnée à afficher par type de milieu (nombre de ramassages trop faible)"
+                )
 
         # Ligne 3 : Graphe par milieu , lieu et année
-        st.write("**Détail par milieu, lieu ou année**")
+        st.write("**Détail par année, type de milieu ou de lieu**")
 
         # Étape 1: Création des filtres
 
-        df_other_metrics = df_other_metrics_raw.copy()
-        df_other_metrics = df_other_metrics.fillna(0)
+        with st.expander("Filtrer par année, type milieu ou type de lieu"):
 
-        selected_annee = st.selectbox(
-            "Choisir une année:",
-            options=["Aucune sélection"] + annee_liste,
-        )
-        if selected_annee != "Aucune sélection":
-            filtered_data_milieu = df_other[df_other["ANNEE"] == selected_annee].copy()
-            filtered_metrics_milieu = df_other_metrics[
-                df_other_metrics["ANNEE"] == selected_annee
-            ].copy()
-        else:
-            filtered_data_milieu = df_other.copy()
-            filtered_metrics_milieu = df_other_metrics.copy()
+            # Filtre par Année
+            # Default values for filters
+            valeur_par_defaut_annee = "Toute la période"
+            valeur_par_defaut_milieu = "Tous les milieux"
+            valeur_par_defaut_lieu = "Tous les lieux"
 
-        selected_type_milieu = st.selectbox(
-            "Choisir un type de milieu:",
-            options=["Aucune sélection"]
-            + list(filtered_data_milieu["TYPE_MILIEU"].unique()),
-        )
+            # Filter by year
```

Comment:
Faudra voir si l'on garde les commentaires en français ou en anglais. Ça fait bizarre de passer de l'un à l'autre.

Reply:
Yes you're right. J'ai tout passé en français